src/Engine/Graphics/Image.cpp

```diff
@@ -57,10 +59,8 @@ const GrayscaleImage &GraphicsImage::indexed() {
     return _indexedImage;
 }
 
-std::string *GraphicsImage::GetName() {
-    assert(_loader);
-
-    return _loader->GetResourceNamePtr();
+const std::string &GraphicsImage::GetName() {
+    return _loader ? _loader->GetResourceName() : globalEmptyString;
```

Comment:
Is there any need to log an error here on empty loader?

Reply:
This is actually perfectly normal - e.g. an automap texture has an empty loader.